File: ShanksBot.java
import java.math.*;
import java.util.*;

public class ShanksBot 
{
    
    public static boolean isPrime(int num) {
        if (num <= 1) return false;
        if (num <= 3) return true;
        if (num % 2 == 0 || num % 3 == 0) return false;
        for (int i = 5; i * i <= num; i += 6) {
            if (num % i == 0 || num % (i + 2) == 0)
                return false;
        }
        return true;
    }
    
    public static int countRepeatingDecimalLength(int denom, int precision) {
        BigDecimal numerator = BigDecimal.ONE;
        BigDecimal denominator = new BigDecimal(denom);
        MathContext mc = new MathContext(precision);
        String result = numerator.divide(denominator, mc).toString();
        String[] parts = result.split("\\.");
        if (parts.length < 2) return 0;
        String decimalPart = parts[1];
        
        for (int length = 1; length <= decimalPart.length() / 2; length++) {
            String pattern = decimalPart.substring(0, length);
            boolean isRepeating = true;
            for (int start = length; start + length <= decimalPart.length(); start += length) {
                if (!decimalPart.substring(start, start + length).equals(pattern)) {
                    isRepeating = false;
                    break;
                }
            }
            if (isRepeating) {
                return length;
            }
        }
        return 0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a prime number:           ");
        int prime = sc.nextInt();
        
        if (!isPrime(prime)) {
            System.out.println(prime + " is not a prime number. Exiting.");
            return;
        }
        
        // Choose a precision high enough for the repeating cycle; using denom * 2 for demonstration.
        int precision = prime * 2;
        
        long startTime = System.nanoTime();
        int repeatingLength = countRepeatingDecimalLength(prime, precision);
        long endTime = System.nanoTime();
        
        double durationInSeconds = (endTime - startTime) / 1_000_000_000.0;
        
        System.out.println("Prime number:                   " + prime);
        System.out.println("Precision used:                 " + precision);
        System.out.println("Length of repeating part:       " + repeatingLength);
        System.out.println("Calculation execution time:     " + durationInSeconds + " seconds");
    }
}

File: ShanksBot.py
import time
from decimal import Decimal, getcontext

def is_prime(num):
    if num <= 1:
        return False
    if num <= 3:
        return True
    if num % 2 == 0 or num % 3 == 0:
        return False
    i = 5
    while i * i <= num:
        if num % i == 0 or num % (i + 2) == 0:
            return False
        i += 6
    return True

def count_repeating_decimal_length(denom, precision):
    getcontext().prec = precision
    decimal_part = str(Decimal(1) / Decimal(denom)).split('.')[-1]
    for length in range(1, len(decimal_part) // 2 + 1):
        pattern = decimal_part[:length]
        if all(decimal_part[i:i+length] == pattern for i in range(length, len(decimal_part) - length, length)):
            return length
    return 0

def main():
    prime = int(input("Enter a prime number:           "))
    if not is_prime(prime):
        print(f"{prime} is not a prime number. Exiting.")
        return
    precision = prime * 2
    start_time = time.time()
    repeating_length = count_repeating_decimal_length(prime, precision)
    duration_in_seconds = time.time() - start_time
    print(f"Prime number:                   {prime}")
    print(f"Precision used:                 {precision}")
    print(f"Length of repeating part:       {repeating_length}")
    print(f"Calculation execution time:     {duration_in_seconds} seconds")

if __name__ == "__main__":
    main()
